File: Core/Entities/Book.cs
using CSharpVitamins;

namespace Core.Entities;

public class Book
{
    public string ISBN { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public string ImageUrl { get; set; }
    public int? Pages { get; set; }
    public decimal Price { get; set; }
    public DateOnly PublishDate { get; set; }
    public Publisher Publisher { get; set; }
    public ShortGuid PublisherId { get; set; }
    public ICollection<Genre> Genres { get; set; } = [];
    public ICollection<Author> Authors { get; set; } = [];
}

File: client/src/app/shared/models/book.ts
import {IAuthor} from "./author";
import {IGenre} from "./genre";

export interface IBook {
  id: number;
  ISBN: string;
  title: string;
  description: string;
  imageUrl: string;
  pages: number;
  price: number;
  publishDate: Date;
  publisher?: string;
  genres: IGenre[];
  authors: IAuthor[];
}
